govcd/metadata_test.go

```diff
@@ -0,0 +1,181 @@
+	if vcd.config.VCD.Vdc == "" {
+		check.Skip("skipping test because no vDC name is empty")
+	}
+
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	// Add metadata
+	task, err := vcd.vdc.AddMetadata("key", "value")
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+	check.Assert(task.Task.Status, Equals, "success")
+
+	// Check if metadata was added correctly
+	metadata, err := vcd.vdc.GetMetadata()
+	check.Assert(err, IsNil)
+	check.Assert(metadata.MetadataEntry[0].Key, Equals, "key")
+	check.Assert(metadata.MetadataEntry[0].TypedValue.Value, Equals, "value")
+}
+
+func (vcd *TestVCD) Test_DeleteMetadataForVdc(check *C) {
+	if vcd.config.VCD.Vdc == "" {
+		check.Skip("skipping test because no vDC name is empty")
+	}
+
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	// Add metadata
+	task, err := vcd.vdc.AddMetadata("key2", "value2")
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+	check.Assert(task.Task.Status, Equals, "success")
+
+	// Remove metadata
+	task, err = vcd.vdc.DeleteMetadata("key2")
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+	check.Assert(task.Task.Status, Equals, "success")
+	metadata, err := vcd.vdc.GetMetadata()
+	check.Assert(err, IsNil)
+	for _, k := range metadata.MetadataEntry {
+		if k.Key == "key2" {
+			check.Errorf("metadata.MetadataEntry should not contain key: %s", k)
+		}
+	}
+}
+
+func (vcd *TestVCD) Test_AddMetadataOnVapp(check *C) {
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp was not successfully created at setup")
+	}
+	// Add metadata
+	task, err := vcd.vapp.AddMetadata("key", "value")
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+	check.Assert(task.Task.Status, Equals, "success")
+
+	// Check if metadata was added correctly
+	metadata, err := vcd.vapp.GetMetadata()
+	check.Assert(err, IsNil)
+	check.Assert(metadata.MetadataEntry[0].Key, Equals, "key")
+	check.Assert(metadata.MetadataEntry[0].TypedValue.Value, Equals, "value")
+}
+
+func (vcd *TestVCD) Test_DeleteMetadataOnVapp(check *C) {
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp was not successfully created at setup")
+	}
+	// Add metadata
+	task, err := vcd.vapp.AddMetadata("key2", "value2")
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+	check.Assert(task.Task.Status, Equals, "success")
+
+	// Remove metadata
+	task, err = vcd.vapp.DeleteMetadata("key2")
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+	check.Assert(task.Task.Status, Equals, "success")
+	metadata, err := vcd.vapp.GetMetadata()
+	check.Assert(err, IsNil)
+	for _, k := range metadata.MetadataEntry {
+		if k.Key == "key2" {
+			check.Errorf("metadata.MetadataEntry should not contain key: %s", k)
+		}
+	}
+}
+
+func (vcd *TestVCD) Test_AddMetadataOnVm(check *C) {
```

Comment:
Could it cleanup (same as one Test_AddMetadataOnVDC)

Reply:
it needs separate approach, cause deletion is for vApp

Reply:
Ok. VMs are being deleted during suite run as opposed to vDC (which is quite permanent).